session.go

```diff
@@ -40,7 +40,8 @@ type session struct {
 	toSend [][]byte
 
 	// Mutex for access to toSend.
-	sendMutex sync.Mutex
+	sendMutex   sync.Mutex
+	resendMutex sync.RWMutex
```

Comment:
Additional comment above this as well indicating the reason why this mutex exists in conjunction with the sendMutex and in which order they must be locked to avoid deadlocks